neo/core/container.py

```diff
@@ -562,6 +562,9 @@ def merge(self, other):
 
         Annotations are merged such that only items not present in the current
         annotations are added.
+
+        Note that the `other` object will be in an inconsistent state after the
+        merge operation and should not be used further.
```

Comment:
It's not clear to me in which cases the `other` objects in corrupted after the merge. Is there a way to prevent this? Otherwise I think many users will miss this comment and run into confusion.

Reply:
if we take the example of two `Segments` A and B, each containing some `AnalogSignals`, and we perform
```
A.merge(B)
```
we add all the signals in `B.analogsignals` to `A.analogsignals`, and we change the `signal.segment` attribute of all the signals in `B.analogsignals` to point to `A`.

The question is then what to do with `B.analogsignals`. We could:

1. empty it. This means that `A.merge(B)` has a side-effect on `B`.
2. leave it unchanged. This means that `B.analogsignals` references signals which do not point back to `B`. This is what I mean by "inconsistent".
3. copy `B` before merging. This has a potentially large memory and performance overhead.

I went with option 2. Option 1 would also be ok, but also has its problems. To minimize confusion, we could add an option like `preserve_other` or `copy`, which would give the user the choice of doing either 2 or 3. In that case, what should the default be: copy or not?

Reply:
Thanks for the clarification. I was thinking the object itself  or child data objects would be inconsistent, but yes, for the links this makes more sense. Maybe change the comment to *Note that the `other` object will be linked inconsistently to other Neo objects after the  merge operation and should not be used further.*
I like your suggestion of adding an option for this. I think it would be most user friendly. Intuitively I would expect option 1) to be the default or would even go beyond, as merging of 2 Segments should only result in a single remaining Segment, i.e. Segment B will be removed.

Reply:
ok. I've changed the comment as you suggest.

If you can merge this PR I'll create a new issue for the remaining changes.